engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala

```diff
@@ -539,6 +519,33 @@ object ProgramExecutionSupport {
     }
   }
 
+  /** Convert the result of Enso visualization function to a byte array.
+    *
+    * @param value the result of Enso visualization function
+    * @return either a byte array representing the visualization result or an
+    *         error
+    */
+  @scala.annotation.tailrec
+  private def visualizationResultToBytes(
+    value: AnyRef
+  ): Either[VisualisationException, Array[Byte]] =
+    value match {
+      case text: String =>
+        Right(text.getBytes(StandardCharsets.UTF_8))
+      case text: Text =>
+        Right(text.toString.getBytes(StandardCharsets.UTF_8))
+      case bytes: Array[Byte] =>
+        Right(bytes)
+      case withWarnings: WithWarnings =>
```

Comment:
I'm not sure why the `WithWarnings` object leaks through the function execution. As far as I understood from @JaroslavTulach explanation, we have to live with it for now.

Reply:
There is https://github.com/oracle/graal/pull/4916 - however the point is to always let `WithWarnings` leak out on the engine level.